Journalator/Source/InvestONator.lua

```
@@ -0,0 +1,391 @@
+    }
+  end
+end
+
+---Recalculate purchased totals from Journalator logs
+---@param fromTime number|nil Minimum timestamp to include (defaults to 0 for all time)
+function Journalator.InvestONator.RecalculateAllPurchases(fromTime)
+  fromTime = fromTime or 0
+
+  -- Build a lookup of total spent per itemID from buyer invoices
+  local spentByItemId = {}
+  local invoices = Journalator.Archiving.GetRange(fromTime, "Invoices")
```

Luacheck (1.2.0)

[warning] mutating non-standard global variable 'Journalator'

(W112)